napari/layers/utils/_slice_input.py

```diff
@@ -78,17 +106,41 @@ def data_indices(
             )
 
         slice_world_to_data = world_to_data.set_slice(self.not_displayed)
-        world_pts = [self.point[ax] for ax in self.not_displayed]
+        world_pts = [self.world_slice.point[ax] for ax in self.not_displayed]
+
+        world_margin_left = [
+            self.world_slice.point[ax] - self.world_slice.margin_left[ax]
+            for ax in self.not_displayed
+        ]
+        world_margin_right = [
+            self.world_slice.point[ax] + self.world_slice.margin_right[ax]
+            for ax in self.not_displayed
+        ]
+
         data_pts = slice_world_to_data(world_pts)
+        data_margin_left = slice_world_to_data(world_margin_left)
+        data_margin_right = slice_world_to_data(world_margin_right)
```

Comment:
This seems like a slow way to apply the same transform to 3 coordinates. Is there some way to stack these and just apply it once?